packages/shared/src/components/auth/AccountCompletionModals.tsx

```diff
@@ -0,0 +1,105 @@
+import { useRouter } from 'next/router';
+import React, { ReactElement, useContext } from 'react';
+import { useQuery } from 'react-query';
+import AuthContext from '../../contexts/AuthContext';
+import useProfileForm from '../../hooks/useProfileForm';
+import { useToastNotification } from '../../hooks/useToastNotification';
+import useWindowEvents from '../../hooks/useWindowEvents';
+import {
+  AuthEvent,
+  AuthFlow,
+  ErrorData,
+  ErrorEvent,
+  getKratosError,
+  getKratosFlow,
+  KRATOS_ERROR,
+} from '../../lib/kratos';
+import { StyledModal } from '../modals/StyledModal';
+import IncompleteRegistrationModal from './IncompleteRegistrationModal';
+import {
+  SocialRegistrationForm,
+  SocialRegistrationFormValues,
+} from './SocialRegistrationForm';
+
+function AccountCompletionModals(): ReactElement {
+  const router = useRouter();
+  const { displayToast } = useToastNotification();
+  const { user, shouldShowLogin, logout, refetchBoot } =
+    useContext(AuthContext);
+  const {
+    updateUserProfile,
+    hint,
+    onUpdateHint,
+    isLoading: isProfileUpdateLoading,
+  } = useProfileForm({
+    onSuccess: () => refetchBoot(),
+  });
+  const onUpdateProfile = ({ name, username }: SocialRegistrationFormValues) =>
+    updateUserProfile({ name, username });
+
+  const displayErrorMessage = (text: string) => {
+    displayToast(text);
+    window.history.replaceState(null, '', '/');
+  };
+
+  useQuery(
```

Comment:
Is this not an existing kratos flow call we already have a hook for?

Reply:
I created a new one but it is still on the works. I think I can tweak it to make this one be part of that.